bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractBaseImpl.java

```diff
@@ -228,6 +240,10 @@ public String getHtmlScreenReaderText() {
     @Override
     @Nullable
     public String getDescription() {
+        if (xssApi != null) {
+            logger.info("[AF] XSS API has been used to encode description HTML.");
+            return translate(PN_DESCRIPTION, xssApi.encodeForHTML(description));
```

Comment:
encodeForHTML is necessary only for Sightly, so it is best to use the xssAPI directly within the Sightly file.

For JSON serialization, the Jackson library automatically handles encoding when converting Java objects to JSON.

Reply:
It unfortunately doesn't. The tooltip and description come as the following in json which is causing the break:
```
{
...
tooltip: "<p>Hello</p>",
...
}
```

Reply:
In that case, we should create a custom serializer and handle this at all places, you can check the way DataRefSerializer is implemented.

Reply:
Implemented using Custom Serializer